src/lib/tls/msg_client_hello.cpp

```diff
@@ -513,12 +531,22 @@ Client_Hello_12::Client_Hello_12(Handshake_IO& io,
    if(policy.support_cert_status_message())
       m_data->extensions().add(new Certificate_Status_Request({}, {}));
 
-   auto supported_groups = std::make_unique<Supported_Groups>(policy.key_exchange_groups());
-   if(!supported_groups->ec_groups().empty())
+   // RFC 8446 4.2.3
+   //    The client MUST NOT include these extensions [Supported Elliptic Curves Extension
+   //    and Supported Point Formats Extension] in the ClientHello message if it does not
+   //    propose any ECC cipher suites.
+   auto offered_ciphers = m_data->ciphersuites();
+   if (std::any_of(offered_ciphers.begin(), offered_ciphers.end(), [](uint16_t cipherSuiteCode) {
+      std::optional<Ciphersuite> cs = Ciphersuite::by_id(cipherSuiteCode);
+      return cs.has_value() && (cs.value().ecc_ciphersuite()); }))
       {
-      m_data->extensions().add(new Supported_Point_Formats(policy.use_ecc_point_compression()));
+      auto supported_groups = std::make_unique<Supported_Groups>(policy.key_exchange_groups());
+      if(!supported_groups->ec_groups().empty())
+         {
+         m_data->extensions().add(new Supported_Point_Formats(policy.use_ecc_point_compression()));
+         }
+      m_data->extensions().add(supported_groups.release());
```

Comment:
Nit. Though, no need to recreate the `unique_ptr`.
```suggestion
      m_data->extensions().add(std::move(supported_groups));
```